### DISCUSSION TITLE
rename `generate` to `build`

### DISCUSSION THREAD

**Discussion by meleu** [bashly-framework/bashly#585] [Ideas]
Maybe I'm biased because I'm a Ruby on Rails developer, but I feel like the `generate` subcommand implies code generation.

The process of merging the partial snippets into a single script seems to be better described as `build`.

(I don't want to be obnoxious, then feel free to close this discussion if you don't want to spend time on this)

---

**UPDATE**: `build` was implemented as an alias to `generate` via https://github.com/DannyBen/bashly/pull/590

**Comment by DannyBen** [comment]
> (I don't want to be obnoxious, then feel free to close this discussion if you don't want to spend time on this)

Not at all - I love these discussions, this is how software gets better.

But you said:

>  I feel like the generate subcommand implies code generation.

which is exactly what `bashly generate` is doing.
I am sure you noticed by now that even if you have exactly 0 lines of code in your own partials, there is a massive amount of code being generated.

Also, you know you can do `bashly g` yes?

If anything, perhaps a `build` / `b` alias - but I don't remember if [my command line framework](https://github.com/dannyben/mister_bin) supports this or not.

**Reply by meleu** [reply]
Yeah, I understand that in the first `generate` run bashly generates a good amount of code.

The feeling that the command name is not describing well what I'm doing happens during the coding/testing phase. I found myself running `bashly generate` many times, but bashly is more "building an artifact" (the final script) than generating code.

I understand it's still generating code, but I see the term "build" being used in other tools that do a similar job, then I think it would be more intuitive. (e.g.: `tsc build`, `npm run build`, `make build`)

**Reply by DannyBen** [reply]
Bashly generates code not only in the first run , but always.

And even though the other commands you have mentioned do not normally generate, I can see the appeal in naming the command `build`.

Since I do not want to break existing toolchains, I wanted to see if I can easily do an alias, but my CLI framework does not currently have this option.

How about:

1. Renaming `generate` to `build`
2. Still keeping `generate` with a `[deprecated]` help text, and keep it like this for a few versions?

**Reply by DannyBen** [reply]
*A few moments later....*

On the other hand..... 

If you ask yourself what is bashly, the answer is a bash CLI generator.
I expect a generator to have a `generate` command.

**Reply by DannyBen** [reply]
Its settled:

`bashly generate` stays the main command name, but it will now also have a `build` alias (or even just `b`).

Implemented in #590 thanks to https://github.com/DannyBen/mister_bin/pull/45

**Reply by meleu** [reply]
> **UPDATE**: I typed the message below before reading the message above :sweat_smile: 

What I originally envisaged was:

- keep `generate` with a deprecation notice
- provide the `build` command (same behavior as `generate` but with no deprecation warning)
- remove `generate` in the next major version bump

**However**, your points are also strong (bashly always generates code) and make me believe that just an alias would be the best solution.

Given the fact your current CLI framework doesn't support aliases, I guess that keeping things as they are is the current best approach. :slightly_smiling_face:

**Reply by DannyBen** [reply]
> UPDATE: I typed the message below before reading the message above 😅

Yes, sorry - I got annoyed by my inability to support aliases, so I rectified that... :)

I am ready to release 1.2.7 if you think it is ready.
I think all your features got in, did we miss anything?

**Reply by meleu** [reply]
I'm gonna test #587 and report back in the PR or issue.
